tests/DateTime/ClockTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Bugsnag\Tests\DateTime;
+
+use Bugsnag\DateTime\Clock;
+use Bugsnag\DateTime\ClockInterface;
+use Bugsnag\Tests\TestCase;
+use DateTimeImmutable;
+
+class ClockTest extends TestCase
+{
+    public function testItImplementsClockInterface()
+    {
+        $clock = new Clock();
+
+        $this->assertInstanceOf(ClockInterface::class, $clock);
+    }
+
+    public function testItReturnsADateTimeImmutable()
+    {
+        $clock = new Clock();
+
+        $this->assertInstanceOf(DateTimeImmutable::class, $clock->now());
+    }
+
+    public function testItReturnsTheCurrentDateTime()
+    {
+        $clock = new Clock();
+
+        $before = new DateTimeImmutable();
+
+        $now = $clock->now();
+
+        $after = new DateTimeImmutable();
```

Comment:
Is this actually getting values aside from all three being equal? Could pop a `usleep` in here to avoid the equal cases.

Reply:
They _should_ all be distinct because `new DateTimeImmutable` includes microseconds, so it's unlikely they'll be the exact same but I let it happen to avoid flakes. Locally they are never the same but maybe with a super PC you could make all 3 dates in the same microsecond :p

I guess a super short `usleep` should be enough to ensure it doesn't ever flake and also not have any noticeable impact on the test length

The other test _needs_ to `usleep` because `Date::now()` only has milliseconds, so technically is before `$before` e.g.

```php
$before = new DateTimeImmutable();
$now = new DateTimeImmutable(Bugsnag\DateTime\Date::now());

var_dump($before, $now);
var_dump($before < $now);
```

```
object(DateTimeImmutable)#3 (3) {
  ["date"]=>
  string(26) "2020-10-26 13:56:23.649783"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(3) "UTC"
}
object(DateTimeImmutable)#2 (3) {
  ["date"]=>
  string(26) "2020-10-26 13:56:23.649000"
  ["timezone_type"]=>
  int(1)
  ["timezone"]=>
  string(6) "+00:00"
}
bool(false)
```

We could construct `$before` without microseconds, but we'd either still have to `usleep` or allow them to be equal

Reply:
Ah! That was actually more complex than I realized 🤔 💭

Reply:
PHP! 🐘